### DISCUSSION TITLE
Mehrere Bridges verbinden >150 Geräte

### DISCUSSION THREAD

**Discussion by mbrockeu** [brutella/hkknx-public#253] [Ideas]
Ich bin mit meiner Installation mittlerweile über die 150 Geräte/Sensoren gekommen die eine Bridge zulässt.

Ich habe daher eine zweite Instanz gestartet und die Geräte etwas aufgeteilt. Jetzt würde ich ja beide Bridges hinzufügen und alles sollte Problemlos laufen. (Konnte ich leider noch nicht testen weil gerade verbaut wird)

Jetzt wäre es aber so, dass jede Bridge eine eigene Verbindung zum IP Interface aufmacht und somit eine Adresse belegt. Kann man vielleicht irgendwie die Verbindung über eine Main Bridge machen, so dass nur eine Verbindung zum IP Interface notwendig ist?

Ist jetzt vielleicht nicht unbedingt der Use-Case für viele, aber vielleicht mal ein Überlegung wert.

**Comment by mirkolenz** [comment]
Ich finde das einen sehr sinnvollen Vorschlag! Hatte auch schon mehrfach überlegt, ein solches Feature anzuregen.

**Comment by mbrockeu** [comment]
Da ich gerade mal wieder in das Problem mit den offenen Tunneln gekommen bin wollte ich mal nachfragen ob man das Thema ggfs. angehen könnte?

**Comment by brutella** [comment]
Weil diese Frage immer wieder aufkommt, würde ich gerne mehr über euer Setup wissen.

- Wieviele Geräte wollt ich insgesamt mit hkknx verwalten?
- Würde es nicht Sinn machen, nur gewisse Geräte mit HomeKit zu verwalten und den Rest mit der neuen Weboberfläche in Version 3?

**Reply by mbrockeu** [reply]
Also aktuell habe ich ca. 190 Geräte und da sind noch nicht wirklich alle drin.
Habe diese nun unterteilt zwischen Schaltbare Geräte und Sensoren.

Das Problem ist, dass man bei einigen Automationen eben auch Sensordaten haben möchte und ggfs. auch am Handy mal schauen, dass ist für mich zumindest aktuell das Hauptgerät womit ich auf mein Smarthome zugreife. Ein Tablet habe ich aktuell nicht. Und alles was auf dem Tablet sein würde, wäre auch automatisch auf dem Handy mehr oder weniger notwendig und im Zweifel sonst über Siri. 

Wie gesagt, Sensoren sind nicht primär notwendig, eben vielleicht nur solche welche man für Automationen bräuchte.
Aktuell stört mich der Aufbau nicht unbedingt, da ich mittlerweile eh 2 IP Gateways hätte. Einfacher wäre vermutlich eher die Verwaltung in einer Anwendung, welche aber mehrere Bridges erstellen kann welche eingebunden werden. Wie die Verteilung auf die Bridges passiert ist eigentlich relativ egal. 

Letzteres wäre für ein Webinterface durchaus sinnvoller, da du ja im Zweifel dort alle Geräte zusammen hast, was bei mehreren Bridges bzw. Instanzen schwierig wird.

Ich habe aktuell übrigens noch HomeAssistant im Einsatz und dort auch ein kleinen Teil in KNX eingebunden. Primär nutz ich das Teil aber als Energiemonitor weil HomeKit das einfach nicht kann.

Hoffe das hilft etwas

**Reply by brutella** [reply]
> Wie die Verteilung auf die Bridges passiert ist eigentlich relativ egal.

Für den Benutzer muss es klar sein, dass es im Hintergrund zwei Bridges gibt und welche Geräte welcher Bridge zugewiesen sind. Ansonsten kann es zur Verwirrung kommen.

> Ich habe aktuell übrigens noch HomeAssistant im Einsatz und dort auch ein kleinen Teil in KNX eingebunden. Primär nutz ich das Teil aber als Energiemonitor weil HomeKit das einfach nicht kann.

Mich würde interessieren, welche Daten und Geräte du für das Energiemonitoring nutzt? (Energiemonitoring soll in Zukunft auch mit Diagrammen in hkknx möglich sein. Dafür muss ich wissen, welche Daten wie aufbereitet werden müssen.)

**Reply by mbrockeu** [reply]
> > Wie die Verteilung auf die Bridges passiert ist eigentlich relativ egal.
> 
> Für den Benutzer muss es klar sein, dass es im Hintergrund zwei Bridges gibt und welche Geräte welcher Bridge zugewiesen sind. Ansonsten kann es zur Verwirrung kommen.

Ich meinte damit auch eher, ob man vorher irgendwo Bridges definiert und die dann zuordnet.
> 
> > Ich habe aktuell übrigens noch HomeAssistant im Einsatz und dort auch ein kleinen Teil in KNX eingebunden. Primär nutz ich das Teil aber als Energiemonitor weil HomeKit das einfach nicht kann.
> 
> Mich würde interessieren, welche Daten und Geräte du für das Energiemonitoring nutzt? (Energiemonitoring soll in Zukunft auch mit Diagrammen in hkknx möglich sein. Dafür muss ich wissen, welche Daten wie aufbereitet werden müssen.)

Das nutze ich aktuell:
![image](https://github.com/brutella/hkknx-public/assets/16849873/56f97589-030e-41d6-a86d-db8d4281ce72)

ich bekomme die Daten vom WR oder von den jeweiligen Autoren und Zeige diese dann entsprechend an. Schön wäre in dem Zusammenhang Gruppierungen in den Bubbles, zum Beispiel ist bei mir bei Solar mehrere drin. Ich kann aber nicht einzeln sehen was kommt. Ist sicher für den Normalnutzer egal weil er nur sehen möchte es kommt Überschuss. Bei den individuellen ist eher die Frage was man hat.

Ich persönlich möchte eher den aktuellen Verbrauch (W) sehen statt den Verbrauch des Tages (kWh).

**Reply by brutella** [reply]
Sendest du diese Daten auf den KNX-Bus und greifst sie dann von dort ab, oder läuft das alles über eine andere Schnittstelle?

**Reply by mbrockeu** [reply]
Das was vom Aktor kommt sende ich natürlich direkt auf den Bus. Das was vom Wechselrichter/Wärmepumpe kommt greife ich über den X1 via Modbus ab und schreibe es dann auf den Bus. Mittlerweile auch direkt in HomeAssistant.

Ich habe aber auch auf dem X1 noch Logiken bzgl. Ladezeitpunkt etc. abhängig von Forecast etc. daher brauch ich das da noch. Passiert aktuell alle 30s. 

Das was die Autoren liefern steht glaube auf 10% Änderung.

**Reply by brutella** [reply]
Danke für die Info. 

In Zukunft möchte ich auch nocht weitere Protokolle (zB Modbus) in Skripte einbauen. Kannst du dir vorstellen, das Auslesen der Werte über ein Skript in hkknx zu probieren, sobald Modbus zur Verfügung steht?

**Comment by gregrahn** [comment]
I have also hit the 149/150 device limit on my hkknx bridge as well.  Any plans on a solution?
Happy to provide any info on the devices, etc. if that is useful for use case analysis, etc.

**Reply by mbrockeu** [reply]
Current workaround is to setup a second bridge with the same key. 
and of course you need a second tunneling possibility 🙈
